### PR TITLE
Remove custom base64 code.

### LINKED ISSUE #2698: Flush SRP's base64 code
Use the generic encoder/decoder.  But see the comments in https://github.com/openssl/openssl/pull/2672

### PR DESCRIPTION
**Pull request by richsalz**
Use EVP_EncodeBlock/EVP_DecodeBlock
Fixes #2698

### CODE REVIEW

File: crypto/srp/srp_vfy.c [hunk 1]
```diff
@@ -19,126 +19,20 @@
...
-    int i, j;
-    int size;
-
-    if (alen == 0 || alen > INT_MAX)
-        return -1;
-
-    while (*src && (*src == ' ' || *src == '\t' || *src == '\n'))
-        ++src;
-    size = strlen(src);
-    if (size < 0 || size >= (int)alen)
```
Comment (paulidale) [inline crypto/srp/srp_vfy.c]: Is the loss of the destination length check safe?

Not that this check is correct.

Comment (paulidale) [inline crypto/srp/srp_vfy.c]: Correction, the check is correct for the old code which converts to six bit completely before reassembling.

File: crypto/srp/srp_vfy.c [hunk 2]
```diff
@@ -19,126 +19,20 @@
...
-        if (--i < 0)
-            break;
-    }
-    while (j <= size && a[j] == 0)
-        ++j;
-    i = 0;
-    while (j <= size)
-        a[i++] = a[j++];
-    return i;
+    return EVP_DecodeBlock(a, (unsigned char *)src, (int)strlen(src));
```
Comment (paulidale) [inline crypto/srp/srp_vfy.c]: The length check would go something like:
```C
size_t size = strlen(src);

if (alen == 0 || alen > INT_MAX || size > INT_MAX || (size / 4) * 3 > alen)
    return -1;
```

Comment (richsalz) [inline crypto/srp/srp_vfy.c]: updated the commit;  the alen tests aren't necessary, given how it is called

File: crypto/srp/srp_vfy.c [hunk 3]
```diff
@@ -19,126 +19,26 @@
...
 static int t_fromb64(unsigned char *a, size_t alen, const char *src)
 {
-    char *loc;
-    int i, j;
-    int size;
+    size_t size = strlen(src);
 
-    if (alen == 0 || alen > INT_MAX)
+    /* Four bytes in src become three bytes output. */
+    if (size > INT_MAX || (size / 4) * 3 > alen);
```
Comment (levitte) [inline crypto/srp/srp_vfy.c]: That semi doesn't look right...

Comment (paulidale) [inline crypto/srp/srp_vfy.c]: I must be blind :(
Good catch.

Comment (levitte) [inline crypto/srp/srp_vfy.c]: I was blind too, but the failing Travis jobs alerted me

Comment (richsalz) [inline crypto/srp/srp_vfy.c]: I found it too. :)  Updated commit pushed.

**Comment by richsalz** [comment]
needs a review now that it's fixed :)

**Comment by richsalz** [comment]
merged, thanks (especially for the careful proof-reading :)

**Pull request closed without merging**